src/gt4py/next/program_processors/runners/dace_fieldview/gtir_builtin_translators.py

```diff
@@ -756,6 +858,298 @@ def translate_scalar_expr(
+    for edge in input_edges:
+        edge.connect(map_entry=None)
+
+    # connect the dataflow result nodes to the carry variables
+    def connect_scan_output(
+        scan_output_edge: gtir_dataflow.DataflowOutputEdge, sym: gtir.Sym
+    ) -> FieldopData:
+        scan_result = scan_output_edge.result
+        assert isinstance(scan_result.gt_dtype, ts.ScalarType)
+        assert scan_result.gt_dtype == sym.type
+        scan_result_data = scan_result.dc_node.data
+        scan_result_desc = scan_result.dc_node.desc(nsdfg)
+
+        output, _ = nsdfg.add_array(
+            scan_output_name(sym.id), scan_output_shape, scan_result_desc.dtype, find_new_name=True
+        )
+        output_node = compute_state.add_access(output)
+        output_subset = str(dace.symbolic.SymExpr(scan_loop_var) - scan_lower_bound)
+        compute_state.add_nedge(
+            scan_result.dc_node, output_node, dace.Memlet(data=output, subset=output_subset)
+        )
+
+        update_state.add_nedge(
+            update_state.add_access(scan_result_data),
+            update_state.add_access(sym.id),
+            dace.Memlet(data=sym.id, subset="0"),
+        )
+
+        output_type = ts.FieldType(dims=[scan_dim], dtype=scan_result.gt_dtype)
+        return FieldopData(output_node, output_type, scan_output_offset)
+
+    lambda_output = (
+        gtx_utils.tree_map(connect_scan_output)(result, scan_carry_input)
+        if (isinstance(result, tuple) and isinstance(scan_carry_input, tuple))
+        else connect_scan_output(result, scan_carry_input)
+        if (
+            isinstance(result, gtir_dataflow.DataflowOutputEdge)
+            and isinstance(scan_carry_input, gtir.Sym)
+        )
+        else None
+    )
+    assert lambda_output
+
+    # in case tuples are passed as argument, isolated non-transient nodes might be left in the state,
+    # because not all tuple fields are necessarily accessed in the lambda scope
+    for data_node in compute_state.data_nodes():
+        data_desc = data_node.desc(nsdfg)
+        if (compute_state.degree(data_node) == 0) and (
+            (not data_desc.transient)
+            or data_node.data.startswith(
+                scan_carry
+            )  # exceptional case where the carry variable is not used, not a scan indeed
+        ):
+            # isolated node, connect it to a transient to avoid SDFG validation errors
+            temp, temp_desc = nsdfg.add_temp_transient_like(data_desc)
+            temp_node = compute_state.add_access(temp)
+            compute_state.add_nedge(data_node, temp_node, dace.Memlet.from_array(temp, temp_desc))
+
+    # build the mapping of symbols from nested SDFG to parent SDFG
+    nsdfg_symbols_mapping: dict[str, dace.symbolic.SymExpr] = {}
+    for dim, _, _ in domain:
+        if dim != scan_dim:
+            dim_map_variable = dace_gtir_utils.get_map_variable(dim)
+            nsdfg_symbols_mapping[dim_map_variable] = dim_map_variable
```

Comment:
You have a type error here, because `dim_map_variable` must be a `str`, but at the same time it must also be a `SymExpr`.